src/Bicep.Core/TypeSystem/Providers/ThirdParty/ThirdPartyResourceTypeProvider.cs

```diff
@@ -0,0 +1,208 @@
+
+                        bodyType = SetBicepResourceProperties(bodyObjectType, resourceType.ValidParentScopes, resourceType.TypeReference, flags);
+                        break;
+                    }
+
+                    bodyType = SetBicepResourceProperties(bodyObjectType, resourceType.ValidParentScopes, resourceType.TypeReference, flags);
+                    break;
+                default:
+                    // we exhaustively test deserialization of every resource type during CI, and this happens in a deterministic fashion,
+                    // so this exception should never occur in the released product
+                    throw new ArgumentException($"Resource {resourceType.TypeReference.FormatName()} has unexpected body type {bodyType.GetType()}");
+            }
+
+            return resourceType with { Body = bodyType };
+        }
+
+        private static ObjectType SetBicepResourceProperties(ObjectType objectType, ResourceScope validParentScopes, ResourceTypeReference typeReference, ResourceTypeGenerationFlags flags)
+        {
+            // Local function.
+            static TypeProperty UpdateFlags(TypeProperty typeProperty, TypePropertyFlags flags) =>
+                new(typeProperty.Name, typeProperty.TypeReference, flags, typeProperty.Description);
+
+            var properties = objectType.Properties;
+            var isExistingResource = flags.HasFlag(ResourceTypeGenerationFlags.ExistingResource);
+
+            if (isExistingResource)
+            {
+            }
+            else
+            {
+                // TODO: remove 'dependsOn' from the type library
+                properties = properties.SetItem(LanguageConstants.ResourceDependsOnPropertyName, new TypeProperty(LanguageConstants.ResourceDependsOnPropertyName, LanguageConstants.ResourceOrResourceCollectionRefArray, TypePropertyFlags.WriteOnly | TypePropertyFlags.DisallowAny));
+            }
+
+            // add the loop variant flag to the name property (if it exists)
+            if (properties.TryGetValue(NamePropertyName, out var nameProperty))
+            {
+                // TODO apply this to all unique properties
+                properties = properties.SetItem(NamePropertyName, UpdateFlags(nameProperty, nameProperty.Flags | TypePropertyFlags.LoopVariant));
+            }
+
+            return new ObjectType(
+                objectType.Name,
+                objectType.ValidationFlags,
+                isExistingResource ? ConvertToReadOnly(properties.Values) : properties.Values,
+                objectType.AdditionalPropertiesType,
+                isExistingResource ? ConvertToReadOnly(objectType.AdditionalPropertiesFlags) : objectType.AdditionalPropertiesFlags,
+                functions: null);
+        }
+
+        private static IEnumerable<TypeProperty> ConvertToReadOnly(IEnumerable<TypeProperty> properties)
+        {
+            foreach (var property in properties)
```

Comment:
This may also require a change to types.json. For now, I think we can just ignore it.